### PR TITLE
Add unit tests for com.jayway.jsonpath.internal.Utils

### PR DESCRIPTION
**Pull request by EricHetti**
I've analysed your codebase and noticed that `com.jayway.jsonpath.internal.Utils` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

File: json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java [hunk 1]
```diff
@@ -1,15 +1,168 @@
...
+        Assert.assertEquals(-1, Utils.indexOf("bar", "foo", 0));
+        Assert.assertEquals(-1, Utils.indexOf("bar", "a", 2));
+        Assert.assertEquals(1, Utils.indexOf("bar", "a", 0));
+        Assert.assertEquals(1, Utils.indexOf("bar", "a", 1));
+    }
+
+    @Test
+    public void testNotNull() {
+        Assert.assertEquals("", Utils.notNull("", "bar", "a", "b", "c"));
+        Assert.assertEquals("foo", Utils.notNull("foo", "bar", "a", "b", "c"));
```
Comment (petersear) [inline json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java:95]: It looks like we can cover another line by passing a first argument of `null`, thus throwing an exception.

File: json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java [hunk 2]
```diff
@@ -1,15 +1,168 @@
...
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
 
     @Test
-    public void strings_can_be_escaped() {
+    public void testJoin() {
+        ArrayList<String> strings = new ArrayList<String>();
+        strings.add("foo");
+        strings.add("bar");
+        strings.add("baz");
```
Comment (ThomasPerkins1123) [inline json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java:20]: Need a break after this line

File: json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java [hunk 3]
```diff
@@ -1,15 +1,168 @@
...
+    @Test
+    public void testIsTruethrows() {
+        thrown.expect(IllegalArgumentException.class);
+        Utils.isTrue(false, "foo");
+    }
+
+    @Test
+    public void testOnlyOneIsTrue() {
+        Utils.onlyOneIsTrue("foo", true, false);
+        Utils.onlyOneIsTrue("foo", false, true);
```
Comment (ThomasPerkins1123) [inline json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java]: is anything actually being tested here other than the methods run without throwing an exception?

**Review by petersear — CHANGES_REQUESTED** [review]

File: json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java [hunk 4]
```diff
@@ -1,15 +1,168 @@
...
+        Assert.assertFalse(Utils.onlyOneIsTrueNonThrow(true, true, true));
+        Assert.assertFalse(Utils.onlyOneIsTrueNonThrow(true, true, false));
+        Assert.assertFalse(Utils.onlyOneIsTrueNonThrow(false, false, false));
+    }
+
+    @Test
+    public void testNotEmpty() {
+        Assert.assertEquals("bar", Utils.notEmpty("bar", "foo", 1, 2, 3));
+        Assert.assertEquals("baz",
+                Utils.notEmpty("baz", "bar", "b", "a", "r"));
```
Comment (ThomasPerkins1123) [inline json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java:146]: break after the first `)` as the `);` should be on its own line:P

**Review by ThomasPerkins1123 — CHANGES_REQUESTED** [review]

File: json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java (hunk 3 above)
Comment (Braavos6) [reply json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java]: Yes.. its just checking if the method is run without throwing an exception

File: json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java (hunk 1 above)
Comment (Braavos6) [reply json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java:95]: Thank you.. Something else was covering that and I missed it

File: json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java (hunk 2 above)
Comment (Braavos6) [reply json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java:20]: Thank you

**Review by petersear — APPROVED** [review]

File: json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java (hunk 3 above)
Comment (ThomasPerkins1123) [reply json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java]: Okay yeah i just had a look at it and it looks like that is all it does so there isn't really much you can do with that is there.

Comment (Braavos6) [reply json-path/src/test/java/com/jayway/jsonpath/internal/UtilsTest.java]: test method removed

**Review by paulkennethkent — APPROVED** [review]

**Review by ThomasPerkins1123 — APPROVED** [review]

**Pull request closed without merging**